### PR TITLE
Fix case sensitivity issue

### PR DESCRIPTION
**Pull request by stillystonka**
Doneness:
- [* ] Code looks good, makes sense, and is high quality
- [* ] Testing added (either as part of this PR or created a related ticket)
- [ ] Documentation modified (if applicable)

### CODE REVIEW

File: phonebuddies/DatabaseConnector.py [hunk 1]
```diff
@@ -111,7 +111,11 @@ def get_all_buddies_map(self):
         buddy_email_map = {}
 
         for buddy in self.get_all_buddies():
-            buddy_email_map[buddy.email] = buddy
+            buddy_email_map[buddy.email.lower()] = buddy  # Converts buddy.email to lower before adding to map
```
Comment (mokolodi1) [inline phonebuddies/DatabaseConnector.py:114]: Do you need to do this given it should be `lower`ed when ingested from the database?

File: phonebuddies/DatabaseConnector.py [hunk 2]
```diff
@@ -111,7 +111,11 @@ def get_all_buddies_map(self):
...
+
+        # this is original, which is case-sensitive
+        # for buddy in self.get_all_buddies():
+        #    buddy_email_map[buddy.email] = buddy
```
Comment (mokolodi1) [inline phonebuddies/DatabaseConnector.py:118]: Would you mind removing this? It's useful to keep around while working, but once I merge, it won't make sense to future contributors.

File: phonebuddies/DatabaseConnector.py [hunk 3]
```diff
@@ -156,7 +160,14 @@ def _find_labeled_row_text(label_text):
...
         topic = _find_labeled_row_text("Topic")
         return EmailInfo(intro, topic)
-    
+
+    def is_email_in_database(self, email):
+        # Get the map of all buddies with their emails as keys
+        buddy_map = self.get_all_buddies_map()
+
+        # Check if the email exists in the buddy_map, considering it case-insensitively
+        return email.lower() in buddy_map
```
Comment (mokolodi1) [inline phonebuddies/DatabaseConnector.py:169]: While this is theoretically useful, I think it's best we only keep the functions we are currently using for now.

File: phonebuddies/Buddy.py [hunk 4]
```diff
@@ -18,10 +18,12 @@ def _parse_row_attribute(self, attribute_name, row_index, required=True, default
         value = None
         if len(self._raw_row_info) >= row_index + 1:
             new_value = self._raw_row_info[row_index]
-
```
Comment (mokolodi1) [inline phonebuddies/Buddy.py:21]: Please keep this whitespace! It helps readability a lot

File: phonebuddies/DatabaseConnector.py [hunk 5]
```diff
@@ -83,7 +83,7 @@ def get_buddy_email_pairs(self):
             if not reject_pair:
                 valid_pairs.append(pair)
 
-        return pairs
+        return valid_pairs
```
Comment (mokolodi1) [inline phonebuddies/DatabaseConnector.py:86]: I think you fixed a bug here that I didn't even know existed 😅

**Review by mokolodi1 — CHANGES_REQUESTED** [review]
Confirmed tests pass! Great work. There's a few nitpick-y things I've left comments on that I'd like cleaned up before I merge, but otherwise this looks good.

File: phonebuddies/DatabaseConnector.py (hunk 5 above)
Comment (stillystonka) [reply phonebuddies/DatabaseConnector.py:86]: GPT caught it haha

**Review by mokolodi1 — APPROVED** [review]
LGTM

**Pull request merged**